spatial/curve/hilbert.go

```diff
@@ -0,0 +1,254 @@
+// Hilbert Curves
+//
+// The Hilbert curve is a continuous, space filling curve first described by
+// David Hilbert. The implementation of Hilbert2D is based on example code from
+// the Wikipedia article
+// (https://en.wikipedia.org/w/index.php?title=Hilbert_curve&oldid=1011599190).
+// The implementation of Hilbert3D and Hilbert4D are extrapolated from
+// Hilbert2D.
+//
+// For the first-order k-dimensional Hilbert curve, a spatial point V is mapped
+// to a point on the curve D by XOR - each dimension of V is expected to be 0 or
+// 1:
+//
+//     func map1stOrder(k int, v []int) (d int) {
+//         for i := k - 1; i >= 0; i-- {
+//             d = d<<1 | (d^v[i])&1
+//         }
+//         return d
+//     }
+//
+// In a 2-space with the origin at the bottom left, this results in a ⊐ shape,
+// wound counter clockwise.
```

Comment:
Please place this in the same comment that documents the package (currently in curve.go, but better in doc.go).